components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2870,4 +2878,59 @@ void RewardsServiceImpl::OnDeleteActivityInfo(
   }
 }
 
+void RewardsServiceImpl::Reset(ResetStateCallback callback) {
+  notification_service_->DeleteAllNotifications();
+  bat_ledger_->DeleteStateFiles(
+      base::BindOnce(&RewardsServiceImpl::OnReset,
+        AsWeakPtr(), std::move(callback)));
+}
+
+void RewardsServiceImpl::OnReset(
+    ResetStateCallback callback,
+    bool success) {
+  std::move(callback).Run(success);
+  for (auto& observer_ : observers_) {
+    observer_.OnReset(this);
+  }
+}
+
+void RewardsServiceImpl::DeleteClientStateFiles(
```

Comment:
as mentioned in the other comment, we can't put the responsibility for knowing what needs to be cleared/deleted on the LedgerClient. The LedgerClient methods are just to perform IO and other operations that the Ledger library is not allowed to perform. We already have save/load/reset state methods and that is what should be used for clearing the state.